Applications/Custom/OptimizerClient/main.cpp

```diff
@@ -29,5 +68,71 @@ int main() {
     return 1;
   }
 
+  auto model =
+    ml::train::createModel(ml::train::ModelType::NEURAL_NET, {"batch_size=10"});
+
+  std::shared_ptr<ml::train::Dataset> dataset;
+  std::shared_ptr<ml::train::Optimizer> optimizer;
+  std::vector<std::shared_ptr<ml::train::Layer>> layers;
+
+  try {
+    dataset = ml::train::createDataset(ml::train::DatasetType::GENERATOR,
```

Comment:
how about using our random data generator as the dataset here?

Reply:
This is not opened for the API, for now, let's stick with it. It will be a good first issue.